src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -33,118 +87,112 @@ LinearIR::constExprIt InsertSpecificIterations::insert_copy_loop(LinearIR& linea
+                                         const std::shared_ptr<op::LoopEnd>& loop_end) {
+    const auto loop_id = loop_end->get_id();
+    const auto& loop_manager = linear_ir.get_loop_manager();
+    const auto& unified_loop_info = loop_manager->get_loop_info<UnifiedLoopInfo>(loop_id);
+
+    auto remaining_work_amount = unified_loop_info->get_work_amount();
+    const auto is_wa_dynamic = utils::is_dynamic_value(remaining_work_amount);
+
+    static constexpr std::array<SpecificLoopIterType, 3> loop_iterations = {
+        SpecificLoopIterType::FIRST_ITER, SpecificLoopIterType::MAIN_BODY, SpecificLoopIterType::LAST_ITER
+    };
+
+    auto decomposed = false;
+    for (const auto& iter_type : loop_iterations) {
+        if (is_decomposed_loop_needed(unified_loop_info, iter_type, remaining_work_amount)) {
+            const auto work_amount = get_decomposed_loop_work_amount(unified_loop_info, iter_type, remaining_work_amount);
+            const auto increment = get_decomposed_loop_increment(unified_loop_info, iter_type, remaining_work_amount);
+            // Update remaining Loop work amount
+            // Note: if work_amount is unknown and increment = 1, it means that a loop will iterate by whole work_amount
+            if ((!is_wa_dynamic) || (increment == 1 && utils::is_dynamic_value(work_amount))) {
+                remaining_work_amount -= work_amount;
+            }
+
+            auto decomposed_loop_end = loop_end;
+            auto decomposed_loop_begin_it = begin, decomposed_loop_end_it = end;
+            auto decomposed_loop_entry_ports = unified_loop_info->get_entry_points();
+            auto decomposed_loop_exit_ports = unified_loop_info->get_exit_points();
+            // Need to copy body if there are other specific sup-loops
+            // Otherwise we should update the current body
+            if (remaining_work_amount > 0) {
+                std::tie(decomposed_loop_begin_it, decomposed_loop_end_it) =
+                    insert_copy_loop(linear_ir, loop_id, begin, decomposed_loop_entry_ports, decomposed_loop_exit_ports);
+                decomposed_loop_end = ov::as_type_ptr<op::LoopEnd>(decomposed_loop_end_it->get()->get_node());
+                OPENVINO_ASSERT(decomposed_loop_end, "Cloned Loop does not contain LoopEnd op at the expected place.");
+
+                // Only latest loop iterations must have summarized finalization offsets!
+                // Since we inserted copy before the latest iterations, these copies should have reseted offsets
+                auto nullify_finalization_offset = [](LoopPort& port) {
+                    if (!utils::is_dynamic_value(port.finalization_offset))
+                        port.finalization_offset = 0;
+                };
+                std::for_each(decomposed_loop_entry_ports.begin(), decomposed_loop_entry_ports.end(), nullify_finalization_offset);
```

Comment:
Minor: We've created a set of special helpers to iterate over loop ports in loop_info.hpp. It looks like a perfect job for such a helper. Why they are not used here?